HKMP/Game/Client/PlayerManager.cs

```diff
@@ -175,29 +275,77 @@ internal class PlayerManager {
         }
 
-        // TODO: investigate whether it is better to disable/setActive(false) player objects instead of destroying
-        // and only destroy when player left server
         /// <summary>
-        /// Destroy the player with the given ID.
+        /// Recycle the player container of the player with the given ID back into the queue.
         /// </summary>
-        /// <param name="id">The player ID.</param>
-        public void DestroyPlayer(ushort id) {
+        /// <param name="id">The ID of the player.</param>
+        public void RecyclePlayer(ushort id) {
             if (!_playerData.TryGetValue(id, out var playerData)) {
-                Logger.Get().Warn(this, $"Tried to destroy player that does not exists for ID {id}");
+                Logger.Get().Warn(this, $"Tried to recycle player that does not exists for ID {id}");
+                return;
+            }
+
+            // First reset the player
+            ResetPlayer(id);
+
+            // Recycle gameObject
+            playerData.PlayerContainer = null;
+            if (!_activePlayers.TryGetValue(id, out var container)) {
+                Logger.Get().Error(this, $"Failed to get a container for player {id} from the active containers.");
                 return;
             }
 
-            // Destroy gameObject and remove from mapping
-            Object.Destroy(playerData.PlayerContainer);
+            container.SetActive(false);
+            container.name = "Player Container";
+
+            _activePlayers.Remove(id);
+            _inactivePlayers.Enqueue(container);
         }
 
         /// <summary>
-        /// Destroy all existing players.
+        /// Recycle all existing players. <seealso cref="RecyclePlayer"/>
         /// </summary>
-        public void DestroyAllPlayers() {
-            foreach (var playerData in _playerData.Values) {
-                // Destroy gameObject
-                Object.Destroy(playerData.PlayerContainer);
+        public void RecycleAllPlayers() {
+            foreach (var id in _playerData.Keys) {
+                // Recycle player
+                RecyclePlayer(id);
+            }
+        }
+
+        /// <summary>
+        /// Reset the player container of the player with the given ID to its initial state by removing all
+        /// game objects not inherent to it.
+        /// </summary>
+        /// <param name="id">The ID of the player.</param>
+        private void ResetPlayer(ushort id) {
+            if (!_playerData.TryGetValue(id, out var playerData)) {
+                Logger.Get().Warn(this, $"Tried to reset player that does not exists for ID {id}");
+                return;
+            }
+
+            var container = playerData.PlayerContainer;
+            if (container == null) {
+                return;
+            }
+
+            // Destroy all descendants and components that weren't originally on the container object,
+            // i.e. added via add-ons
```

Comment:
This comment is not (yet) accurate.
```suggestion
            // Destroy all descendants and components that weren't originally on the container object
```